pkg/storegateway/bucket.go

```diff
@@ -1144,6 +1151,7 @@ func (s *BucketStore) getSeriesIteratorFromBlocks(
 
 	stats.update(func(stats *queryStats) {
 		stats.blocksQueried = len(batches)
+		stats.setBlocksQueriedBySourceAndLevel(blocksQueriedBySourceAndLevel)
```

Comment:
maybe I'm overly fixating on this or it's still early morning, but to prevent creating and resizing additional maps I'd just iterate over the blocks within `stats.update` and update `blocksQueriedBySourceAndLevel` with each block. not feeling strongly about this.

Reply:
Yea, the reason I didn't do that is only because we'd need to initialize the map before writing to it, and presently the Stats object is initialized in a few different places. We could centralize these into a constructor method. Any preference?

Reply:
there's already `newSafeQueryStats` which is used throughout for safe stats. The unsafe stats aren't used  We _may_ be able to change only that?

Reply:
There are a couple other places where `queryStats` is initialized, such as [here](https://github.com/grafana/mimir/blob/main/pkg/storegateway/series_refs.go#L894). Any preference on approach?

Reply:
I added a `new` method to initialize these. Let me know if you'd like something different.